tests/quantization/test_bitsandbytes.py

```diff
@@ -5,82 +5,108 @@
+    with vllm_runner(model_name,
+                     quantization='bitsandbytes',
+                     load_format='bitsandbytes',
+                     enforce_eager=True) as llm:
+        model = llm.model.llm_engine.model_executor.driver_worker.model_runner.model  # noqa: E501
+
+        # check the weights in MLP & SelfAttention are quantized to torch.int8
+        validate_model_weight_type(model, torch.int8)
+
+        validate_model_output(llm)
+
+
+def validate_model_weight_type(model, quantized_dtype=torch.uint8):
+    # Check quantized weights
+    quantized_layers = [('mlp.gate_up_proj.qweight',
+                         model.model.layers[0].mlp.gate_up_proj.qweight),
+                        ('mlp.down_proj.qweight',
+                         model.model.layers[0].mlp.down_proj.qweight),
+                        ('self_attn.o_proj.qweight',
+                         model.model.layers[0].self_attn.o_proj.qweight),
+                        ('self_attn.qkv_proj.qweight',
+                         model.model.layers[0].self_attn.qkv_proj.qweight)]
+
+    for name, qweight in quantized_layers:
+        assert qweight.dtype == quantized_dtype, (
+            f'Expected {name} dtype {quantized_dtype} but got {qweight.dtype}')
+
+    # Check non-quantized weights
+    non_quantized_layers = [
+        ('lm_head.weight', model.lm_head.weight),
+        ('embed_tokens.weight', model.model.embed_tokens.weight),
+        ('input_layernorm.weight',
+         model.model.layers[0].input_layernorm.weight),
+        ('post_attention_layernorm.weight',
+         model.model.layers[0].post_attention_layernorm.weight)
+    ]
+
+    for name, weight in non_quantized_layers:
+        assert weight.dtype != quantized_dtype, (
+            f'{name} dtype should not be {quantized_dtype}')
+
+
+def validate_model_output(llm: VllmRunner):
+    sampling_params = SamplingParams(temperature=0.0,
+                                     logprobs=1,
+                                     prompt_logprobs=1,
+                                     max_tokens=8)
+
+    prompts = ['That which does not kill us', 'To be or not to be,']
+    expected_outputs = [
+        'That which does not kill us makes us stronger.',
+        'To be or not to be, that is the question.'
+    ]
+    outputs = llm.generate(prompts, sampling_params=sampling_params)
+    assert len(outputs) == len(prompts)
+
+    for index in range(len(outputs)):
+        # compare the first line of the output
+        actual_output = outputs[index][1][0].split('\n', 1)[0]
+        expected_output = expected_outputs[index].split('\n', 1)[0]
+
+        assert len(actual_output) >= len(expected_output), (
+            f'Actual {actual_output} should be larger than or equal to '
+            f'expected {expected_output}')
+        actual_output = actual_output[:len(expected_output)]
+
+        assert actual_output == expected_output, (
+            f'Expected: {expected_output}, but got: {actual_output}')
```

Comment:
Could you change this to do a direct top-logprob greedy comparison between HF and vLLM? There are several references in the testing but here is one https://github.com/vllm-project/vllm/blob/9ba85bc1527928f30ec56961520d7e07ee385167/tests/models/test_mistral.py#L29-L35

Reply:
> Could you change this to do a direct top-logprob greedy comparison between HF and vLLM? There are several references in the testing but here is one

Updated the code per comments, could you take a look?